source/braille.py

```diff
@@ -2234,32 +2246,34 @@ def _get_script(self):
 					gestureKeys -= keys
 		if not gestureModifiers:
 			return None
-		if gestureKeys != set(self.keyNames):
+		if gestureKeys != set(keyNames):
 			# Find a script for L{gestureKeys}.
 			id = "+".join(gestureKeys)
-			fakeGestureIds = [u"br({source}):{id}".format(source=self.source, id=id),]
+			fakeGestureIds = [
+				inputCore.normalizeGestureIdentifier(u"br({source}):{id}".format(source=self.source, id=id)),
+			]
 			if self.model:
-				fakeGestureIds.insert(0,u"br({source}.{model}):{id}".format(source=self.source, model=self.model, id=id))
+				fakeGestureIds.insert(0,
+					inputCore.normalizeGestureIdentifier(u"br({source}.{model}):{id}".format(source=self.source, model=self.model, id=id))
+				)
 			scriptNames = []
+			# We can't bother about multiple scripts for a gesture, we will just use the first one
+			# Therefore, global maps should be processed in most to least important order.
+			# The user gesture map could override assignments in the display gesture map.
 			globalMaps = [inputCore.manager.userGestureMap, handler.display.gestureMap]
 			for globalMap in globalMaps:
 				for fakeGestureId in fakeGestureIds:
-					scriptNames.extend(scriptName for cls, scriptName in globalMap.getScriptsForGesture(fakeGestureId.lower()) if scriptName.startswith("kb"))
+					scriptNames.extend(scriptName for cls, scriptName in globalMap.getScriptsForGesture(fakeGestureId) if scriptName is None or scriptName.startswith("kb"))
 			if not scriptNames:
 				# Gesture contains modifiers, but no keyboard emulate script exists for the gesture without modifiers
 				return None
-			# We can't bother about multiple scripts for a gesture, we will just use the first one
-			combinedScriptName = "kb:{modifiers}+{keys}".format(
-				modifiers="+".join(gestureModifiers),
-				keys=scriptNames[0].split(":")[1]
-			)
-		elif script and scriptName:
-			combinedScriptName = "kb:{modifiers}+{keys}".format(
-				modifiers="+".join(gestureModifiers),
-				keys=scriptName.split(":")[1]
-			)
-		else:
+			scriptName = scriptNames[0]

```

Comment:
Hmm, instead of extending the scriptNames list, I could imagine the creation of a nested loop, i.e.

```
for globalMap in globalMaps:
    for gestureId in gestureIds:
        for cls, scriptName in globalMap.getScriptsForGesture(gestureId):
            if scriptName is something:
                break
```

However as you know, this only breaks out of the inner most loop. I guess the only way to do this, would be by means of a function. I can declare a local function inside _get_script, but don't know whether that is very ideal.